pkg/apis/application/v1alpha1/types.go

```diff
@@ -570,7 +572,8 @@ func (k *ApplicationSourceKustomize) IsZero() bool {
 			len(k.Replicas) == 0 &&
 			len(k.CommonLabels) == 0 &&
 			len(k.CommonAnnotations) == 0 &&
-			len(k.Patches) == 0
+			len(k.Patches) == 0 &&
+			len(k.Components) == 0
```

Comment:
Good catch. Since this touches the filesystem we should also update `AllowsConcurrentProcessing` in this file.

Reply:
@crenshaw-dev fixed